ast/src/input/program_input/main_input.rs

```diff
@@ -62,8 +72,26 @@ impl MainInput {
         Ok(())
     }
 
-    /// Returns an `Option` of the main function input at `name`
-    pub fn get(&self, name: &str) -> Option<Option<InputValue>> {
-        self.input.get(name).cloned()
+    /// Parses constants input definitions and stores them in `self`.
+    pub fn parse_constants(&mut self, definitions: Vec<Definition>) -> Result<(), InputParserError> {
+        for definition in definitions {
+            let name = definition.parameter.variable.value;
+            let value = InputValue::from_expression(definition.parameter.type_, definition.expression)?;
+
+            self.insert_constant(name, Some(value));
+        }
+
+        Ok(())
+    }
+
+    /// Returns an `Option` of the main function input at `name`. As a second
+    /// value in a tuple returns flag whether input is defined as a constant in input file.
+    pub fn get(&self, name: &str) -> Option<(Option<InputValue>, bool)> {
```

Comment:
When does this method get called?
Would it be better to split it up into two methods?

Reply:
MainInput holds both inputs and constant inputs, so I decided to update it to not break compatibility. Technically, they are both inputs, and `get()` as well as `len()` and `empty()` handles both 'types' of inputs.